src/site/content/en/blog/codelab-sign-up-form-best-practices/index.md

```diff
@@ -0,0 +1,269 @@
+  </section>
+
+  <section>        
+    <label for="email">Email</label>
+    <input id="email" name="email" autocomplete="username"
+           type="email" required>
+  </section>
+  
+  <section>
+    <label for="password">Password</label>
+    <input id="password" name="password" autocomplete="new-password" 
+           type="password" minlength="8" required>
+  </section>
+  
+  <button id="sign-up">Sign up</button>
+  
+</form>
+```
+
+The `type` values do a lot:
+* `type="password"` obscures text as it's entered and enables the browser's 
+[password manager](https://passwords.google.com/) to suggest a strong password. 
+* `type="email"` provides basic validation and ensures mobile users get an appropriate keyboard.
+
+Try submitting the form with an empty field. The browser won't submit the form, and it prompts to 
+complete missing data and sets focus. That's because you added the `require` attribute to all the 
+inputs. Now try submitting with a password that has less than eight characters. The browser warns that 
+the password isn't long enough and sets focus on the password input. The same works for `pattern` 
+(used for the name input) and other [validation constraints](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5/Constraint_validation). The browser does all this automatically, without needing any extra code.
+
+Using the `autocomplete` value `name` for the **Full name** input makes sense, but what about the 
+other inputs? 
+* `autocomplete="username"` for the **Email** input means the browser's password manager will store 
+the email address as the 'name' for this user (the username!) to go with the password.
+* `autocomplete="new-password"` for **Password** is a hint to the password manager that it should 
+offer to store this value as the password for the current site. You can then use 
+`autocomplete="current-password"` to enable autofill in a sign-in form (remember, this is 
+*sign-up* form). 
+
+{% Aside %}
+[Sign-in form best practices](/sign-in-form-best-practices) has more tips for improving form design, 
+layout and accessibility.
+{% endAside %}
+
+
+## Step 4: Help users enter secure passwords 
+
+With the form as it is, did you notice anything wrong with the password input?
+
+Two problems:
+* There's no way to know if there are constraints on the password value.
+* You can't see the password to check if you got it right.
+
+Don't make users guess!
+
+Update the password section of `index.html`:
+
+```html
+<section>
+  <label for="password">Password</label>
+  <button id="toggle-password" type="button" aria-label="Show password as plain text. 
+          Warning: this will display your password on the screen.">Show password</button>
+  <input id="password" name="password" type="password" autocomplete="new-password" 
+         minlength="8" aria-describedby="password-constraints" required>
+  <div id="password-constraints">Eight or more characters.</div>
+</section>
+```
+
```

Comment:
```suggestion

This adds new features to help users enter passwords:

* A button (actually just text) to toggle password display. (The button functionality will be added with JavaScript.)
* An `aria-label` attribute for the password-toggle button. 
* An `aria-describedby` attribute for the password input. Screenreaders read the label text, the 
input type (password), and then the description. 
```

I moved this up because I find it easier to follow if the code is explained step by step. (And I reordered the bullet points in order of appearance.)

Reply:
Thanks.